### PR TITLE
Add option to deploy locally through Docker

### PR DESCRIPTION
**Pull request by richardlin047**
This PR enables DataSpread to be deployed locally through Docker. This reduces the required software and number of setup steps the user needs to get DataSpread up and running.

This does not change any logic of DataSpread, but `/dataspread-ui/pom.xml` is modified to use a different plugin so that Node can be installed within the docker container.

Configuration files for Tomcat are added to the `build-web` folder. The Docker deployment uses scripts from `docker-compose.yml` and the `Dockerfile` in `/` and `/build-db`. 

Rebuilding the Docker image can take a while since maven dependencies are not cached. This would be a good next step to reduce build times.

### CODE REVIEW

**Comment by twattanawaroon** [comment]
I tried running the Docker on my linux machine and it seems to work fine. I'm a bit unsure about the postgres version req.

**Comment by richardlin047** [comment]
> I tried running the Docker on my linux machine and it seems to work fine. I'm a bit unsure about the postgres version req.

I only made a change to fix a typo in the postgres requirement. Should I change or remove the postgres version requirement?

File: build-db/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Use postgres 9.5
+FROM postgres:9.5
```
Comment (twattanawaroon) [inline build-db/Dockerfile:2]: I was referring to here. I believe we actually use some geometric functionalities from 10+.

Comment (richardlin047) [reply build-db/Dockerfile]: Updated image to 10.14.

**Pull request merged**